### ISSUE TITLE
High CPU usage when extensions are disabled 

### ISSUE THREAD

**Issue by kaznovac** [docker/for-mac#6310]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 736DF04E-A73B-4F61-A3B0-0925DD507977/20220510122220

### Expected behavior
low cpu usage

### Actual behavior
`/Applications/Docker.app/Contents/MacOS/com.docker.extensions` uses near 100
`/Applications/Docker.app/Contents/MacOS/com.docker.backend` uses near 80

### Information

  - macOS Version: 12.3.1
  - Intel chip or Apple chip: Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
  - Docker Desktop Version: 4.8.1 (78998)

### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check`
```
% /Applications/Docker.app/Contents/MacOS/com.docker.diagnose check
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0031: does the Docker API work?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0001: is the application running?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0017: can a VM be started?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0007: is the backend responding?
[PASS] DD0014: are the backend processes running?
[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[FAIL] DD0012: is the VM networking working? network checks failed: failed to ping host: exit status 1
[2022-05-10T12:26:21.048179000Z][com.docker.diagnose][I] ipc.NewClient: 3547ca77-diagnose-network -> Library/Containers/com.docker.docker/Data/diagnosticd.sock diagnosticsd
[common/pkg/diagkit/gather/diagnose.runIsVMNetworkingOK()
[	common/pkg/diagkit/gather/diagnose/network.go:34 +0xdd
[common/pkg/diagkit/gather/diagnose.(*test).GetResult(0x4d10180)
[	common/pkg/diagkit/gather/diagnose/test.go:46 +0x43
[common/pkg/diagkit/gather/diagnose.Run.func1(0x4d10180)
[	common/pkg/diagkit/gather/diagnose/run.go:17 +0x5a
[common/pkg/diagkit/gather/diagnose.walkOnce.func1(0x2?, 0x4d10180)
[	common/pkg/diagkit/gather/diagnose/run.go:140 +0x77
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x1, 0x4d10180, 0xc00067b728)
[	common/pkg/diagkit/gather/diagnose/run.go:146 +0x36
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x0, 0x4?, 0xc00067b728)
[	common/pkg/diagkit/gather/diagnose/run.go:149 +0x73
[common/pkg/diagkit/gather/diagnose.walkOnce(0x46d9700?, 0xc00035f888)
[	common/pkg/diagkit/gather/diagnose/run.go:135 +0xcc
[common/pkg/diagkit/gather/diagnose.Run(0x4d10300, 0x10?, {0xc00035fb18, 0x1, 0x1})
[	common/pkg/diagkit/gather/diagnose/run.go:16 +0x1cb
[main.checkCmd({0xc000032050?, 0x6?, 0x4?}, {0x0, 0x0})
[	common/cmd/com.docker.diagnose/main.go:132 +0x105
[main.main()
[	common/cmd/com.docker.diagnose/main.go:98 +0x2a8
[2022-05-10T12:26:21.048438000Z][com.docker.diagnose][I] (dce739c5) 3547ca77-diagnose-network C->S diagnosticsd POST /check-network-connectivity: {"ips":["10.0.3.107"]}
[2022-05-10T12:26:21.561736000Z][com.docker.diagnose][W] (dce739c5) 3547ca77-diagnose-network C<-S 8e4bd001-diagnosticsd POST /check-network-connectivity (513.333163ms): failed to ping host: exit status 1

[PASS] DD0032: do Docker networks overlap with host IPs?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0019: is the com.docker.vmnetd process responding?
[PASS] DD0033: does the host have Internet access?

Please investigate the following 1 issue:

1 : The test: is the VM networking working?
    Failed with: network checks failed: failed to ping host: exit status 1

VM seems to have a network connectivity issue. Please check your host firewall and anti-virus settings in case they are blocking the VM.

```

although reported the network actually works

### Steps to reproduce the behavior

  1. disable extensions
  2. wait

**Comment by caseywebdev** [comment]
There is also a memory leak with extensions disabled. Both of the processes listed above rose well past the 16GB limit I'd set, eventually reaching over 80GB before I shut it down. Running with extensions enabled today and they seem stable.

**Comment by afedotov** [comment]
I'm observing the same issue.
BTW, there is an intense log flood during high cpu usage. This host log files:
- ~/Library/Containers/com.docker.docker/Data/log/host/com.docker.backend.log
- ~/Library/Containers/com.docker.docker/Data/log/host/com.docker.extensions.log

**Comment by actualben** [comment]
I'm also seeing the high cpu usage and extreme memory leak with extensions disabled. This is after a factory reset followed by deleting the app, then downloading the latest version to achieve a completely fresh install of 4.8.1 (78998) macOS 12.3.1 (21E258).  `com.docker.extensions` and `com.docker.backend` are producing about 9,800 log messages per second in the system log.

<img width="762" alt="Screenshot 2022-05-10 at 10 51 20 PM" src="https://user-images.githubusercontent.com/10159132/167719824-cf82fcf5-9d8d-4887-9b12-fa52c58743af.png">

**Comment by gtardif** [comment]
Thx for the feedback, we're looking into this issue. 
> Running with extensions enabled today and they seem stable.

Thx, good to see this works as a workaround until we fix the issue

**Comment by timothyallan** [comment]
11.6.5 here. When Idle, there's a constant cranked CPU and a nonstop flood of log files and killed mds services on 4.8.1

I'd disabled extensions previously in a beta version I got from a previous GH issue :|

![Screen Shot 2022-05-10 at 9 55 32 PM](https://user-images.githubusercontent.com/1995135/167766233-239f3e47-a98f-48c4-ba5e-98df5f12a636.png)
![Screen Shot 2022-05-10 at 9 54 39 PM](https://user-images.githubusercontent.com/1995135/167766240-e43ed6ad-183b-4a05-aac0-f40beb7e1796.png)
1

**Comment by kaznovac** [comment]
high memory usage seems to be present regardless of the extension enabled state,
while high cpu usage remains reserved to disabled extensions

**Comment by lorenrh** [comment]
We are preparing a 4.8.2 release. We have an [RC mac Intel build](https://desktop.docker.com/mac/main/amd64/78998/Docker.dmg) and [RC mac ARM build](https://desktop.docker.com/mac/main/arm64/78998/Docker.dmg) prepared. Would you mind checking it out?

If the issue still persists, could you [create diagnostics](https://docs.docker.com/desktop/mac/troubleshoot/#diagnose-and-feedback) and post back the ID once uploaded, please?

**Comment by mat007** [comment]
Hi, we have just released Docker Desktop 4.8.2 with a fix. Thanks!

**Comment by mat007** [comment]
If anyone is still experiencing a memory leak with Desktop 4.8.2, could you please open a new issue?
Thank you!
